pkg/controller/podautoscaler/pod_filters.go

```diff
@@ -207,7 +220,12 @@ func (f *OwnerReferencesFilter) isTargetMatch(obj *unstructured.Unstructured, ta
 
 // isOwnerRefMatch checks if an owner reference matches the target reference
 func (f *OwnerReferencesFilter) isOwnerRefMatch(ownerRef metav1.OwnerReference, targetRef autoscalingv2.CrossVersionObjectReference) bool {
-	return ownerRef.Kind == targetRef.Kind && ownerRef.Name == targetRef.Name && ownerRef.APIVersion == targetRef.APIVersion
+	kindMatch := ownerRef.Kind == targetRef.Kind
+	nameMatch := ownerRef.Name == targetRef.Name
```

Comment:
I don't know how to correctly handle this, but in the VPA we have this logic, and it didn't work with the targetRef was `deployment`, since the lower case `d` didn't match the Kind of Deployment.
I don' know if the answer is to `ToLower()`, or something else.